### PR TITLE
Removing feature flag for Embedded KYC

### LINKED ISSUE #9397: Remove Feature flag for Embedded KYC
### Description

In [#9592
](https://github.com/Automattic/woocommerce-payments/issues/9252) we added a feature flag to the embedded KYC. This issue is to revert that PR when we're ready to release Embedded KYC.

### Acceptance criteria

* Roll back the changes made in https://github.com/Automattic/woocommerce-payments/pull/9394 so that embedded KYC is the default option.
* Clean up any code that is no longer being used now that embedded KYC is the default.

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #9397 

#### Changes proposed in this Pull Request

* Remove the feature flag gating the Embedded KYC feature
* Some code cleanup

#### TODO

- [x] Merge the PR before next release
- [x] Make a new issue for further code cleanup (e.g. we can probably remove a lot of the code that dealt with the old method of redirecting to external KYC, but I'd suggest to wait until this feature has been out there for a few months first.

#### Testing instructions

* Start a new onboarding and confirm it uses the embedded KYC.
* Test the onboarding (including leaving the onboarding midway through embedded KYC and then coming back to it) and verify you are able to successfully onboard an account.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9445` or branch name `dev/remove-feature-flag` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/remove-feature-flag) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9ddd8cec9537f262321222e005834424cfb88538
- Build time: 2024-09-18 09:00:27 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/index.js [hunk 1]
```diff
@@ -70,25 +70,19 @@ addFilter(
 			capability: 'manage_woocommerce',
 		} );
 
-		// Currently under feature flag.
-		if (
-			wcpaySettings &&
-			wcpaySettings.featureFlags.isEmbeddedKycEnabled
```
Comment (oaratovskyi) [inline client/index.js:76]: nitpick: you should also remove `isEmbeddedKycEnabled` from `globals.d.ts`

File: tests/unit/test-class-wc-payments-account.php [hunk 2]
```diff
@@ -843,8 +843,6 @@ public function test_maybe_handle_onboarding_init_embedded_kyc() {
 			->expects( $this->never() )
 			->method( 'redirect_to_onboarding_wizard' );
 
-		update_option( '_wcpay_feature_embedded_kyc', '1' );
```
Comment (oaratovskyi) [inline tests/unit/test-class-wc-payments-account.php:846]: nitpick: you should also delete this from `tests/unit/test-class-wc-payments-account.php` - `tear_down` function

**Review by oaratovskyi — APPROVED** [review]
Looks good and works well! 🥇 
Approving, left a few nitpick comments, nothing blocking!

File: tests/unit/test-class-wc-payments-account.php (hunk 2 above)
Comment (oaratovskyi) [reply tests/unit/test-class-wc-payments-account.php:846]: Done in https://github.com/Automattic/woocommerce-payments/pull/9445/commits/7ca58a8f8f51d5df09e4a187d5cf270f73396667

File: client/index.js (hunk 1 above)
Comment (oaratovskyi) [reply client/index.js:76]: Done in https://github.com/Automattic/woocommerce-payments/pull/9445/commits/7ca58a8f8f51d5df09e4a187d5cf270f73396667

**Comment by oaratovskyi** [comment]
Make a new issue for further code cleanup (e.g. we can probably remove a lot of the code that dealt with the old method of redirecting to external KYC, but I'd suggest to wait until this feature has been out there for a few months first.

Issue created https://github.com/Automattic/woocommerce-payments/issues/9457

**Pull request merged**